src/RuleHandlerResolver/RuleHandlerContainer.php

```diff
@@ -11,18 +11,53 @@
 use Yiisoft\Validator\RuleHandlerInterface;
 use Yiisoft\Validator\RuleHandlerResolverInterface;
 
+/**
+ * An implementation for {@see RuleHandlerResolverInterface} acting as a wrapper over dependency injection container
+ * ({@see ContainerInterface}) throwing more specific exceptions and executing some additional checks (to make sure that
+ * if a handler was found, then it's indeed a valid handler to work with) during resolving a rule handler name.
+ *
+ * To use it, make sure to change `config.php` like so:
+ *
+ * ```php
+ * use Yiisoft\Validator\RuleHandlerResolverInterface;
+ * use Yiisoft\Validator\RuleHandlerResolver\RuleHandlerContainer;
+ *
+ * [
+ *     RuleHandlerResolverInterface::class => RuleHandlerContainer::class,
+ * ];
+ * ```
+ *
+ * If you don't need DI container, {@see SimpleRuleHandlerContainer} can be used instead. It's enabled by default to you
+ * don't need to additionally configure anything.
+ */
 final class RuleHandlerContainer implements RuleHandlerResolverInterface
 {
-    public function __construct(private ContainerInterface $container)
-    {
+    public function __construct(
+        /**
+         * @var ContainerInterface An instance of dependency injection container.
+         */
+        private ContainerInterface $container,
+    ) {
     }
 
-    public function resolve(string $className): RuleHandlerInterface
+    /**
+     * Resolves a rule handler name to a corresponding rule handler instance. The actual resolving is delegated to
+     * {@see $container}. Throws more specific exceptions and executes some additional checks (to make sure that if a
+     * handler was found, then it's indeed a valid handler to work with).
+     *
+     * @param string $name A rule handler name ({@see RuleInterface}).
+     *
+     * @throws RuleHandlerInterfaceNotImplementedException if a found instance is not a valid rule handler.
```

Comment:
```suggestion
     * @throws RuleHandlerInterfaceNotImplementedException If a found instance is not a valid rule handler.
```

Reply:
Suggest to leave as is, because it reads like a one sentence (starting with "throws").